### PR TITLE
[Doc] Add 3.3 MV supports Iceberg Partition Transform

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wangsimo0 — DISMISSED** [review]

File: docs/en/using_starrocks/create_partitioned_materialized_view.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ Currently, StarRocks supports building partitioned materialized views on tables
   - Currently, list partitioning and expression partitioning are not supported.
   - The two adjacent partitions of the base table must have consecutive ranges.
 - For multi-level partitioned base tables in external catalogs, only the primary level partitioning path can be used to create a partitioned materialized view. For example,  for a table partitioned in the `yyyyMMdd/hour` format, you can only build the materialized view partitioned by `yyyyMMdd`.
+- From v3.2.3, StarRocks supports creating partitioned materialized views upon Iceberg tables with [Partition Transforms](https://iceberg.apache.org/spec/#partition-transforms), and the materialized views are partitioned by the column after the transform. For more information, see [Data lake query acceleration with materialized views - Choose a suitable refresh strategy](./data_lake_query_acceleration_with_materialized_views.md#choose-a-suitable-refresh-strategy).
```
Comment (evelynzhaojie) [inline docs/en/using_starrocks/create_partitioned_materialized_view.md:51]: after the transformation.

File: docs/en/using_starrocks/data_lake_query_acceleration_with_materialized_views.md [hunk 2]
```diff
@@ -106,6 +106,34 @@ You can still choose to tolerate a certain level of data inconsistency by settin
 
 Please note that if you need to refresh by partition, the partitioning keys of the materialized view must be included in that of the base table.
 
+From v3.2.3, StarRocks supports creating partitioned materialized views upon Iceberg tables with [Partition Transforms](https://iceberg.apache.org/spec/#partition-transforms), and the materialized views are partitioned by the column after the transform. Currently, only Iceberg tables with the `identity`, `year`, `month`, `day`, or `hour` transforms are supported.
```
Comment (evelynzhaojie) [inline docs/en/using_starrocks/data_lake_query_acceleration_with_materialized_views.md:109]: after the transformation.

File: docs/zh/using_starrocks/data_lake_query_acceleration_with_materialized_views.md [hunk 3]
```diff
@@ -105,6 +105,34 @@ StarRocks 支持基于 External Catalog，如 Hive Catalog、Iceberg Catalog、H
 
 请注意，如需按照分区刷新，物化视图的分区键必须包含在基表的分区键中。
 
+从 v3.2.3 版本开始，StarRocks 支持在使用 [Partition Transforms (分区变换)](https://iceberg.apache.org/spec/#partition-transforms) 的 Iceberg 表上创建分区物化视图，物化视图将根据变换后的列进行分区。目前，仅支持使用 `identity`、`year`、`month`、`day` 或 `hour` 变换的 Iceberg 表。
```
Comment (evelynzhaojie) [inline docs/zh/using_starrocks/data_lake_query_acceleration_with_materialized_views.md:108]: 仅支持使用 `identity`、`year`、`month`、`day` 或 `hour` 变换 --- 这里的变换用英文 transform 更容易理解

File: docs/zh/using_starrocks/data_lake_query_acceleration_with_materialized_views.md [hunk 4]
```diff
@@ -105,6 +105,34 @@ StarRocks 支持基于 External Catalog，如 Hive Catalog、Iceberg Catalog、H
...
+
+以下示例展示了具有 `day` 分区变换的 Iceberg 表的定义，并在其上创建了分区对齐的物化视图：
```
Comment (evelynzhaojie) [inline docs/zh/using_starrocks/data_lake_query_acceleration_with_materialized_views.md:110]: ”具有“这个词改写一下

**Review by evelynzhaojie — APPROVED** [review]

**Review by hellolilyliuyi — APPROVED** [review]

**Pull request merged**